elements/drag-drop-elements/pl-drag-drop-element.py

```diff
@@ -0,0 +1,389 @@
+        student_previous_submission = []
+        submission_indent = []
+
+        pl.check_attribs(element,
+                         required_attribs=['answers-name'],
+                         optional_attribs=['shuffle-options',
+                                           'permutation-mode',
+                                           'check-indentation',
+                                           'header-left-column',
+                                           'header-right-column',
+                                           'external-grader',
+                                           'file-name',
+                                           'leading-code',
+                                           'trailing-code',
+                                           'max-distractors',   # Legacy attribute
+                                           'max-feedback-count'])  # Legacy attribute
+
+        for html_tags in element:
+            if html_tags.tag == 'pl-answer':
+                # CORRECT is optional for backward compatibility
+                # pl.check_attribs(html_tags, required_attribs=[''], optional_attribs=['correct', 'ranking', 'indent'])
+                mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+            # BACKWARD COMPATIBILITY CODE
+            if html_tags.tag == 'pl-distractor':
+                mcq_options.append(str.strip(html_tags.text))   # store the original specified ordering of all the MCQ options
+
+        answerName = pl.get_string_attrib(element, 'answers-name')
+        header_left_column = pl.get_string_attrib(element, 'header-left-column', 'Drag from here:')
+        header_right_column = pl.get_string_attrib(element, 'header-right-column', 'Construct your solution here:')
+
+        student_submission_dict_list = []
+
+        mcq_options = data['params'][answerName]
+
+        if answerName in data['submitted_answers']:
+            student_previous_submission = data['submitted_answers'][answerName]['student_raw_submission']
+            mcq_options = list(set(mcq_options) - set(student_previous_submission))
+
+        for index, mcq_options_text in enumerate(student_previous_submission):
+            # render the answers column (restore the student submission)
+            submission_indent = data['submitted_answers'][answerName]['student_answer_indent'][index]
+            submission_indent = (int(submission_indent) * 50) + 5
+            temp = {'text': mcq_options_text, 'indent': submission_indent}
+            student_submission_dict_list.append(dict(temp))
+
+        html_params = {
+            'question': True,
+            'answerName': answerName,
+            'options': mcq_options,
+            'header-left-column': header_left_column,
+            'header-right-column': header_right_column,
+            'submission_dict': student_submission_dict_list
+            # 'data': data
```

Comment:
Shouldn't have commented out code in a pull request.

Reply:
fixed in bcadc9f